### PR TITLE
backport 29233b8 ship

### PR DESCRIPTION
**Pull request by tbfleming**
Backport ship from 29233b8

* Kept ship format at 2.0.x, except `get_status_result_v0` has the additional `chain_id` field
* Kept 2.0.x's fix to shut down on ship write error
* Kept 2.0.x's block_start handling

### CODE REVIEW

File: libraries/state_history/include/eosio/state_history/log.hpp [hunk 1]
```diff
@@ -87,52 +87,39 @@ class state_history_log {
 
    template <typename F>
    void write_entry(const state_history_log_header& header, const chain::block_id_type& prev_id, F write_payload) {
-      try {
```
Comment (swatanabe) [inline libraries/state_history/include/eosio/state_history/log.hpp:90]: This diff is from https://github.com/EOSIO/eos/pull/10332
Did you check this out to make sure that it doesn't need to be ported?

File: libraries/state_history/include/eosio/state_history/serialization.hpp [hunk 2]
```diff
@@ -356,7 +356,7 @@ datastream<ST>& operator<<(datastream<ST>& ds, const history_serial_wrapper<eosi
 
 template <typename ST>
 datastream<ST>& operator<<(datastream<ST>& ds, const history_serial_wrapper<eosio::chain::shared_key_weight>& obj) {
-   fc::raw::pack(ds, as_type<eosio::chain::public_key_type>(obj.obj.key));
+   fc::raw::pack(ds, as_type<eosio::chain::shared_public_key>(obj.obj.key));
```
Comment (swatanabe) [inline libraries/state_history/include/eosio/state_history/serialization.hpp:359]: I'm pretty sure that we need this change: https://github.com/EOSIO/eos/pull/9088

Comment (tbfleming) [reply libraries/state_history/include/eosio/state_history/serialization.hpp]: yes. fixed.

File: libraries/state_history/include/eosio/state_history/log.hpp (hunk 1 above)
Comment (tbfleming) [reply libraries/state_history/include/eosio/state_history/log.hpp:90]: I moved the catch to the plugin's `on_accepted_block` so non-appbase apps (e.g. cltester) can use the state_history lib

File: tests/testUtils.py [hunk 3]
```diff
@@ -73,8 +73,6 @@ class Utils:
     DataDir="var/lib/"
     ConfigDir="etc/eosio/"
 
-    TimeFmt='%Y-%m-%dT%H:%M:%S.%f'
```
[the remark this answers is not in the document]
Comment (tbfleming) [reply tests/testUtils.py]: fixed

**Review by swatanabe — APPROVED** [review]

**Pull request merged**